pandas/tests/plotting/frame/test_frame.py

```diff
@@ -696,6 +696,36 @@ def test_scatterplot_object_data(self):
         _check_plot_works(df.plot.scatter, x="a", y="b")
         _check_plot_works(df.plot.scatter, x=0, y=1)
 
+    @pytest.mark.parametrize("ordered", [True, False])
+    @pytest.mark.parametrize(
+        "categories",
+        (["setosa", "versicolor", "virginica"], ["versicolor", "virginica", "setosa"]),
+    )
+    def test_scatterplot_color_by_categorical(self, ordered, categories):
```

Comment:
Is there a way to add assertions on the colorbar ticklabels for the ordered case?

Reply:
I couldn't find a simple way to do this using the public API from mpl, but can look into it further (or do you have a suggestion?)

Reply:
I look at my comment once again and I cannot figure out what kind of assertions I asked you to consider.
You already do check the ticklabels at the colorbar against the expectations.
Sorry for the noise.